sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectView.groovy

```diff
@@ -106,6 +108,44 @@ class ProjectView extends ProjectDesign {
         headerRow.getCell("SampleDataAvailable").setDescription("Number of available raw datasets.")
     }
 
+    private void addSorting(){
+        sort.setItems(["Recently Updated", "Last Recently Updated", "Subscribed", "Not Subscribed"])
+
+        addSortColumns()
+
+        sort.addValueChangeListener({
+            if(it.value){
+                switch (it.value){
+                    case "Subscribed":
+                        projectGrid.sort("isSubscribed", SortDirection.DESCENDING)
+                        break
+                    case "Not Subscribed":
+                        projectGrid.sort("isSubscribed", SortDirection.ASCENDING)
+                        break
+                    case "Recently Updated":
+                        projectGrid.sort("lastUpdated", SortDirection.DESCENDING)
+                        break
+                    case "Last Recently Updated":
+                        projectGrid.sort("lastUpdated", SortDirection.ASCENDING)
+                        break
+                    default:
+                        projectGrid.clearSortOrder()
+                }
+            }else{
+                projectGrid.clearSortOrder() //FYI because the dataprovider content is sorted by last updated this is how its sorted currently
+            }
+        })
+    }
+
+    /**
+     * Adds hidden columns used for sorting the grid
+     */
+    private void addSortColumns(){
+        projectGrid.addColumn({it.hasSubscription}).setId("isSubscribed").setHidden(true)
+
+        projectGrid.addColumn({it.lastChanged}).setId("lastUpdated").setHidden(true)
```

Comment:
this is fine, though I think you can move this to the general grid setup and only set the comparator here.